stringsubstitutor/src/main/java/org/jdbi/v3/stringsubstitutor/StringSubstitutorTemplateEngineFactory.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jdbi.v3.stringsubstitutor;
+
+/**
+ * Convenient semantic factory for common {@link StringSubstitutorTemplateEngine} instances.
+ */
+public class StringSubstitutorTemplateEngineFactory {
+    private StringSubstitutorTemplateEngineFactory() {}
+
+    public static StringSubstitutorTemplateEngine between(char prefix, char suffix) {
```

Comment:
Let's fold these factory methods into `StringSubstitutorTemplateEngine`, so the usage looks like:

```java
jdbi.setTemplateEngine(StringSubstitutorTemplateEngine.between('<','>'));
```

Reply:
You sure? Feels like interface pollution to me tbh...

Reply:
I tend to agree -- if the factory had any state or configurability, then a separate class is fine.  But as is these are simple static factory methods that should probably go on the class itself.

```java
private StringSubTemplateEngine(Consumer<StringSubstitutor> configurer) { ... }
public static SSTE between(prefix, suffix) {
    return new SSTE(sub -> {
        sub.setVar...
    });
}
```